### PR TITLE
fix `Server shutdown with "Error: Can't set headers after they are sent." ` (close #937)

### LINKED ISSUE #937: Server shutdown with "Error: Can't set headers after they are sent."
Reproducing on https://e.mail.ru/messages/inbox/
The `fatalError` event emits two times.
First stack:
```
    at RequestPipelineContext.closeWithError (d:\Projects\testcafe-hammerhead\lib\request-pipeline\context.js:268:44)
    at error (d:\Projects\testcafe-hammerhead\lib\request-pipeline\index.js:264:85)
    at DestinationRequest.<anonymous> (d:\Projects\testcafe-hammerhead\lib\request-pipeline\index.js:95:24)
    at emitOne (events.js:77:13)
    at DestinationRequest.emit (events.js:169:7)
    at DestinationRequest._onError (d:\Projects\testcafe-hammerhead\lib\request-pipeline\destination-request\index.js:184:46)
    at ClientRequest.<anonymous> (d:\Projects\testcafe-hammerhead\lib\request-pipeline\destination-request\index.js:103:31)
    at emitOne (events.js:77:13)
    at ClientRequest.emit (events.js:169:7)
    at TLSSocket.socketCloseListener (_http_client.js:235:9)
```
Second stack:
```
    at RequestPipelineContext.closeWithError (d:\Projects\testcafe-hammerhead\lib\request-pipeline\context.js:268:44)
    at error (d:\Projects\testcafe-hammerhead\lib\request-pipeline\index.js:264:85)
    at DestinationRequest.<anonymous> (d:\Projects\testcafe-hammerhead\lib\request-pipeline\index.js:95:24)
    at emitOne (events.js:77:13)
    at DestinationRequest.emit (events.js:169:7)
    at DestinationRequest._onTimeout (d:\Projects\testcafe-hammerhead\lib\request-pipeline\destination-request\index.js:176:18)
    at ClientRequest.<anonymous> (d:\Projects\testcafe-hammerhead\lib\request-pipeline\destination-request\index.js:106:31)
    at ClientRequest.g (events.js:260:16)
    at emitNone (events.js:67:13)
    at ClientRequest.emit (events.js:166:7)
```

### PR DESCRIPTION
**Pull request by LavrovArtem**
/cc @churkin @miherlosev

### CODE REVIEW

File: src/request-pipeline/destination-request/index.js [hunk 1]
```diff
@@ -112,8 +112,11 @@ export default class DestinationRequest extends EventEmitter {
             this._send();
         }
 
-        else if (this._isDNSErr(err))
+        else if (this._isDNSErr(err)) {
+            this._abort();
```
Comment (inikulin) [inline src/request-pipeline/destination-request/index.js]: It makes sense to refactor whole code into method:
```js
_fatalError(msg) {
     this.aborted = true;
     this.req.abort();
     this.emit('fatalError', getText(msg, this.opts.url));
}
```

Remove `_abort` and use `_fatalError` everythere

Comment (LavrovArtem) [inline src/request-pipeline/destination-request/index.js]: @inikulin thanks, good idea

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
@testcafe-build-bot \retest

**Comment by LavrovArtem** [comment]
FPR

**Comment by LavrovArtem** [comment]
ping @inikulin

**Pull request merged**